cobigen/cobigen-core-api/src/test/java/com/devonfw/cobigen/api/MavenUtilTest.java

```diff
@@ -0,0 +1,148 @@
+package com.devonfw.cobigen.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.devonfw.cobigen.api.util.MavenUtil;
+import com.devonfw.cobigen.api.util.to.JfrogSearchResponse;
+import com.devonfw.cobigen.api.util.to.MavenSearchResponse;
+import com.devonfw.cobigen.api.util.to.NexusSearchResponse;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Test class for maven utilities
+ */
+public class MavenUtilTest {
+
+  /** Testdata root path */
+  private static final String testdataRoot = "src/test/resources/testdata/unittest/MavenUtilTest";
+
+  /**
+   * Tests if maven json response can properly be parsed and converted to a list of download URLs
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testMavenParseDownloadLinks() throws IOException {
+
+    ObjectMapper mapper = new ObjectMapper();
+    MavenSearchResponse response = new MavenSearchResponse();
+
+    String jsonResponse = new String(Files.readAllBytes(Paths.get(testdataRoot).resolve("mavenJsonTest.json")));
+
+    response = mapper.readValue(jsonResponse, MavenSearchResponse.class);
+    List<URL> downloadLinks = response.getDownloadURLs();
+    assertThat(downloadLinks).contains(
+        new URL("https://repo1.maven.org/maven2/com/google/inject/guice/5.1.0/guice-5.1.0.jar"),
+        new URL("https://repo1.maven.org/maven2/com/google/inject/guice-bom/5.1.0/guice-bom-5.1.0.jar"),
+        new URL("https://repo1.maven.org/maven2/com/google/inject/guice-parent/5.1.0/guice-parent-5.1.0.jar"),
+        new URL("https://repo1.maven.org/maven2/com/google/inject/jdk8-tests/5.0.1/jdk8-tests-5.0.1.jar"));
```

Comment:
Isn't it better to put these strings as constants?

Reply:
Not in this case, as it resembles the "parsed" content of the test resource file.